DB- Old 526 Submission Flow Cleanup

SarahJaine opened the pull request:
<!-- Please read our guidelines before submitting your first PR https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/code_review_guidelines.md -->

## Description of change
<!-- Please include a description of the change and context. What would a code reviewer, or a future dev, need to know about this PR in order to understand why this PR is necessary? This could include dependencies introduced, changes in behavior, pointers to more detailed documentation. The description should be more than a link to an issue.  -->
**Part 2 of 2, fixes [va.gov-team 6822](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6822)**
Remove unused code related to abandoned asynchronous approach to form 526 submissions.
First PR is https://github.com/department-of-veterans-affairs/vets-api/pull/4039, contains `model/` related changes

There is no data in either of the production tables that are removed by this PR.

## Testing
<!-- Please describe testing done to verify the changes or any testing planned. -->

## Acceptance Criteria (Definition of Done)

#### Unique to this PR
<!-- This would be a good place to include feature flag check item and info, specific dashboards and instrumentation check item and info -->
- [x] Remove `DisabilityCompensationSubmission` and `DisabilityCompensationJobStatus` models and their associated test specs and tables
- [x] Schema is updated

#### Applies to all PRs

- [n/a] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [n/a] Provide which alerts would indicate a problem with this functionality (if applicable)

SarahJaine commented on db/migrate/20200317162403_remove_disability_compensation_job_statuses.rb:
```diff
@@ -0,0 +1,5 @@
+class RemoveDisabilityCompensationJobStatuses < ActiveRecord::Migration[5.2]
+  def change
+    drop_table :disability_compensation_job_statuses
```
I don't really like using change here because the automagic down would be to create the table. This doesn't actually reverse the effects of dropping a table because creating the table doesn't reinstate dropped data, dropping a table is in reality irreversible. So, I originally wanted to writeup/down functions instead and have my down raise ActiveRecord::IrreversibleMigration but that kind of stinks too because there isn't any data in this table or the disability_compensation_submissions table in prod-- it seems silly to fail rollbacks for this. In summation, all these solutions are imperfect, so I picked the least controversial way forward.

johnpaulashenfelter replied:
I get the concern and appreciate the thought -- but since there's no table structure here, what do you get when you reverse the `drop_table`. I honestly never thought about it

johnpaulashenfelter replied:
Actually figured it out

```
$ rake db:migrate
Running via Spring preloader in process 30545
/Users/johnpaul/work/oddball/vets-api/rakelib/connectivity.rake:11: warning: already initialized constant REDIS_CONFIG
/Users/johnpaul/work/oddball/vets-api/config/initializers/redis.rb:4: warning: previous definition of REDIS_CONFIG was here
== 20200317162403 RemoveDisabilityCompensationJobStatuses: migrating ==========
-- drop_table(:disability_compensation_job_statuses)
   -> 0.0026s
== 20200317162403 RemoveDisabilityCompensationJobStatuses: migrated (0.0061s) =

== 20200317164309 RemoveDisabilityCompensationSubmissions: migrating ==========
-- drop_table(:disability_compensation_submissions)
   -> 0.0008s
== 20200317164309 RemoveDisabilityCompensationSubmissions: migrated (0.0035s) =

johnpaul@buffy:~/work/oddball/vets-api (sj-remove-VA526ez-vgt6822-part2) $ rake db:rollback
Running via Spring preloader in process 30689
/Users/johnpaul/work/oddball/vets-api/rakelib/connectivity.rake:11: warning: already initialized constant REDIS_CONFIG
/Users/johnpaul/work/oddball/vets-api/config/initializers/redis.rb:4: warning: previous definition of REDIS_CONFIG was here
== 20200317164309 RemoveDisabilityCompensationSubmissions: reverting ==========
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:



To avoid mistakes, drop_table is only reversible if given options or a block (can be empty).
```

SarahJaine replied:
I don't love that it also errors.  Do you want me to to do an explicit up/down? I can make down can have the table creation with the col definitions.

johnpaulashenfelter replied:
I think explicit up/down is easier to read. The alternative is 

```
  def change
    drop_table :disability_compensation_job_statuses, id: :serial, force: :cascade  do |t|
      t.datetime "created_at", null: false
      t.datetime "updated_at", null: false
      t.integer "disability_compensation_id"
      t.integer "va526ez_submit_transaction_id"
      t.boolean "complete", default: false
    end
  end
```
which i think isn't really straightforward to understand (why would a drop_table take a block?.... ooooooohhhhh for a rollback)

SarahJaine commented:
Was finally able to test this rollback/migrate in docker, works!
```
vets-api@7f4d844a7996:~/src$ bin/rails db:rollback
== 20200317164309 RemoveDisabilityCompensationSubmissions: reverting ==========
-- create_table("disability_compensation_submissions", {:id=>:serial, :force=>:cascade})
   -> 0.0084s
== 20200317164309 RemoveDisabilityCompensationSubmissions: reverted (0.0085s) =
vets-api@7f4d844a7996:~/src$ bin/rails db:rollback
== 20200317162403 RemoveDisabilityCompensationJobStatuses: reverting ==========
-- create_table("disability_compensation_job_statuses", {:id=>:serial, :force=>:cascade})
   -> 0.0193s
== 20200317162403 RemoveDisabilityCompensationJobStatuses: reverted (0.0196s) =
vets-api@7f4d844a7996:~/src$ bin/rails db:migrate VERBOSE=true
== 20200317162403 RemoveDisabilityCompensationJobStatuses: migrating ==========
-- drop_table(:disability_compensation_job_statuses)
   -> 0.0029s
== 20200317162403 RemoveDisabilityCompensationJobStatuses: migrated (0.1329s) =

== 20200317164309 RemoveDisabilityCompensationSubmissions: migrating ==========
-- drop_table(:disability_compensation_submissions)
   -> 0.0023s
== 20200317164309 RemoveDisabilityCompensationSubmissions: migrated (0.1321s) =
```
